nengo/tests/test_processes.py

```diff
@@ -334,3 +334,108 @@ def test_present_input(Simulator, rng):
     y = sim.data[up].reshape(len(t), c, ni, nj)
     for k, [ii, image] in enumerate(zip(i, y)):
         assert np.allclose(image, images[ii], rtol=1e-4, atol=1e-7), (k, ii)
+
+
+class TestPiecewise(object):
+
+    dt = 0.001
+
+    def run_sim(self, data, interpolation, Simulator):
+        process = nengo.processes.Piecewise(data, interpolation)
+
+        with nengo.Network() as model:
+            u = nengo.Node(process, size_out=process.default_size_out)
+            up = nengo.Probe(u)
+
+        with Simulator(model, dt=self.dt) as sim:
+            sim.run(1.5)
+
+        return sim.data[up]
+
+    def test_basic(self, Simulator):
+        f = self.run_sim({0.5: 1, 1.0: 0}, 'zero', Simulator)
+        assert np.allclose(f[0], [0])   # t = 0.001
+        assert np.allclose(f[249], [0]) # t = 0.25
+        assert np.allclose(f[498], [0]) # t = 0.499
+        assert np.allclose(f[499], [1]) # t = 0.5
+        assert np.allclose(f[749], [1]) # t = 0.75
+        assert np.allclose(f[999], [0]) # t = 1.0
+        assert np.allclose(f[1499], [0]) # t = 1.5
+
+
+    def test_lists(self, Simulator):
+        f = self.run_sim({0.5: [1, 0], 1.0: [0, 1]}, 'zero', Simulator)
+        assert np.allclose(f[0], [0, 0])   # t = 0.001
+        assert np.allclose(f[249], [0, 0]) # t = 0.25
+        assert np.allclose(f[498], [0, 0]) # t = 0.499
+        assert np.allclose(f[499], [1, 0]) # t = 0.5
+        assert np.allclose(f[749], [1, 0]) # t = 0.75
+        assert np.allclose(f[999], [0, 1]) # t = 1.0
+        assert np.allclose(f[1499], [0, 1]) # t = 1.5
+
+
+
+    def get_step(self, data):
+        process = nengo.processes.Piecewise(data)
+        return process.make_step(shape_in=(process.default_size_in,),
+                                 shape_out=(process.default_size_out,),
+                                 dt=0.001, rng=None)
+
+
+    def test_default_zero(self):
+        f = self.get_step({0.5: 1, 1.0: 0})
+        assert np.allclose(f(-10), [0])
+        assert np.allclose(f(0), [0])
+
+
+    def test_invalid_key(self):
+        data = {0.5: 1, 1: 0, 'a': 0.2}
+        with pytest.raises(ValidationError):
+            process = nengo.processes.Piecewise(data)
+            assert process
+
+
+    def test_invalid_length(self):
+        data = {0.5: [1, 0], 1.0: [1, 0, 0]}
+        with pytest.raises(ValidationError):
+            process = nengo.processes.Piecewise(data)
+            assert process
+
+
+    def test_invalid_interpolation_type(self):
+        data = {0.5: 1, 1.0: 0}
+        with pytest.raises(ValidationError):
+            process = nengo.processes.Piecewise(data, 'not-interpolation')
+            assert process
+
+    def test_invalid_interpolation_dimention(self):
+        data = {0.5: [1, 0], 1.0: [0, 1]}
+        with pytest.raises(ValidationError):
+            process = nengo.processes.Piecewise(data, 'cubic')
+            assert process
+
+    def test_interpolation(self, Simulator):
```

Comment:
At the moment, this test requires scipy, which is not a required dependency of Nengo. In order to not fail unit tests when scipy is not installed, please use `pytest.importorskip` (see [test_dists.py](https://github.com/nengo/nengo/blob/master/nengo/tests/test_dists.py#L176) for an example).